blueprints/component-test/index.js

```diff
@@ -1,61 +1,68 @@
+/*jshint node:true*/
+
+var EOL  = require('os').EOL;
+
 module.exports = {
-  description: 'Generates a component test.',
+  description: 'Generates a component integration or unit test.',
 
   availableOptions: [
     {
-      name: 'integration',
-      type: Boolean,
-      default: false,
-      aliases: ['i']
+      name: 'test-type',
+      type: ['integration', 'unit'],
+      default: 'integration',
+      aliases: [
+        { 'i': 'integration'},
+        { 'u': 'unit'},
+        { 'integration': 'integration' },
+        { 'unit': 'unit' }
+      ]
     }
   ],
 
   locals: function(options) {
-    var additionalImports, prefix, testOpts, defaultTest;
-    if (options.integration) {
-      additionalImports = "import hbs from 'htmlbars-inline-precompile';";
-      prefix = 'Integration: ';
-      testOpts = 'integration: true';
-      defaultTest = "// renders the component on the page\n" +
-             "       this.render(hbs`{{" + options.entity.name + "}}`);\n" +
-             "       expect(this.$()).to.be.ok;";
-    } else {
-      additionalImports = '';
+    var testImports, prefix, testOpts, testContent;
+    if (options.testType === 'unit') {
+      testImports = '';
       prefix = '';
-      testOpts =  "// specify the other units that are required for this test\n" +
+      testOpts =  "// specify the other units that are required for this test" + EOL +
               "    // needs: ['component:foo', 'helper:bar']";
-      defaultTest = "// creates the component instance\n" +
-              "      var component = this.subject();\n" +
-              "      expect(component._state).to.equal('preRender');\n\n" +
-              "      // renders the component on the page\n" +
-              "      this.render();\n" +
+      testContent = "// creates the component instance" + EOL +
+              "      var component = this.subject();" + EOL +
+              "      expect(component._state).to.equal('preRender');" + EOL + EOL +
+              "      // renders the component on the page" + EOL +
+              "      this.render();" + EOL +
               "      expect(component._state).to.equal('inDOM');";
+    } else {
+      testImports = EOL + "import hbs from 'htmlbars-inline-precompile';";
+      prefix = 'Integration: ';
+      testOpts = 'integration: true';
+      testContent = "// Set any properties with this.set('myProperty', 'value');" + EOL +
+              "      // Handle any actions with this.on('myAction', function(val) { ... });" + EOL +
+              "      // Template block usage:" + EOL +
+              "      // this.render(hbs`" + EOL +
+              "      //   {{#" + options.entity.name + "}}" + EOL +
+              "      //     template content" + EOL +
+              "      //   {{/" + options.entity.name + "}}" + EOL +
+              "      // `);" + EOL + EOL +
```

Comment:
The ember-cli version actually uses the block format and tests that it yields appropriately, but I don't want to make that assumption for people.